tests/rptest/services/redpanda.py

```diff
@@ -2465,23 +2481,31 @@ def start_si(self):
             self.logger.debug(
                 f"Creating ABS container: {self._si_settings.cloud_storage_bucket}"
             )
+        else:
+            raise RuntimeError(
+                f"Unsuported cloud_storage_type: {self._si_settings.cloud_storage_type}"
+            )
 
         if not self._si_settings.bypass_bucket_creation:
+            assert self._si_settings.cloud_storage_bucket, "No SI bucket configured"
```

Comment:
A lot of asserts had to be added here, as the type checker can see that these proeprties are sometimes null (when cloud storage isn't configured): some outside force presumably ensures we don't run cloud storage-using tests in that scenario, but the type checker doens't know here. 

Some additional refactoring/abstraction could help here but I just went the easy route.